File: backend/dot-net/Controllers/CommentController.cs
using System.Security.Claims;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using backend.DTOs;
using backend.Services;

namespace backend.Controllers;

[ApiController]
[Route("api/v1/comments")]
public class CommentController : ControllerBase
{
    private readonly CommentService _commentService;

    public CommentController(CommentService commentService)
    {
        _commentService = commentService;
    }

    private string? GetUserId() => User.FindFirstValue("userId");
    private string? GetUsername() => User.FindFirstValue("username");

    [HttpPost]
    [Authorize]
    public IActionResult Create([FromBody] CommentCreateRequest request)
    {
        var comment = _commentService.Create(
            request.Content, request.QuestionId,
            GetUserId()!, GetUsername()!
        );
        return CreatedAtAction(nameof(Create), 
            new { success = true, message = "Comment created successfully", data = comment });
    }

    [HttpGet("question/{questionId}")]
    public IActionResult GetByQuestionId(string questionId)
    {
        var comments = _commentService.GetByQuestionId(questionId);
        return Ok(new { success = true, message = "Success get comments", data = comments });
    }

    [HttpDelete("{id}")]
    [Authorize]
    public IActionResult Delete(string id)
    {
        _commentService.Delete(id);
        return Ok(new { success = true, message = "Comment deleted successfully" });
    }
}

File: backend/node-express/src/controllers/comment.controller.js
const commentService = require('../services/comment.service');

class CommentController {
  async create(req, res, next) {
    try {
      const { content, questionId } = req.body;
      const userId = req.user.userId;

      const comment = await commentService.create({
        content,
        questionId,
        userId,
        username: req.user.username
      });

      res.status(201).json({
        success: true,
        message: 'Comment created successfully',
        data: comment
      });
    } catch (error) {
      next(error);
    }
  }

  async getByQuestionId(req, res, next) {
    try {
      const { questionId } = req.params;
      const comments = await commentService.getByQuestionId(questionId);
      res.json({
        success: true,
        message: 'Success get comments',
        data: comments
      });
    } catch (error) {
      next(error);
    }
  }

  async delete(req, res, next) {
    try {
      const { id } = req.params;
      await commentService.delete(id);
      res.json({
        success: true,
        message: 'Comment deleted successfully'
      });
    } catch (error) {
      next(error);
    }
  }
}

module.exports = new CommentController();
